CommonApi/src/main/java/mezz/jei/api/registration/IAdvancedRegistration.java

```diff
@@ -17,4 +20,18 @@ public interface IAdvancedRegistration {
 	 * Register your own {@link IRecipeManagerPlugin} here.
 	 */
 	void addRecipeManagerPlugin(IRecipeManagerPlugin recipeManagerPlugin);
+
+	/**
+	 * Register a {@link IRecipeCategoryDecorator} for a recipe category.
+	 *
+	 * @since 15.1.0
+	 */
+	<T> void addRecipeCategoryDecorator(IRecipeCategory<T> recipeCategory, IRecipeCategoryDecorator<T> decorator);
```

Comment:
This is the same as calling
`addRecipeCategoryDecorator(recipeCategory.getRecipeType(), decorator)`
so I think we can remove it

Reply:
It was added because adding the decorator to a recipe type requires iteration of all recipe categories in order to validate the type and whether it has a category attached to it.
It may not be that impactful. I'll remove it.

Reply:
Yeah there are not that many recipe categories to iterate over so it's not a big deal, maybe 1000s in a big pack.
For optimal performance we could use a `Map<RecipeType, RecipeCategory>` internally, I'm sure I already create that somewhere else